File: 1.1/src/cz/mg/vulkan/PFNvkDestroyFramebuffer.java
package cz.mg.vulkan;

public class PFNvkDestroyFramebuffer extends VkFunctionPointer {
    public PFNvkDestroyFramebuffer() {
    }

    public PFNvkDestroyFramebuffer(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkDestroyFramebuffer(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroyFramebuffer(long value) {
        setValue(value);
    }

    public PFNvkDestroyFramebuffer(VkInstance instance) {
        super(instance, new VkString("vkDestroyFramebuffer"));
    }

    public void call(VkDevice device, VkFramebuffer framebuffer, VkAllocationCallbacks pAllocator){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), framebuffer != null ? framebuffer.getVkAddress() : VkPointer.getNullAddressNative(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void call(long vkaddress, long device, long framebuffer, long pAllocator);
}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkDestroyFramebuffer.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroyFramebuffer_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong framebuffer, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroyFramebuffer f = (PFN_vkDestroyFramebuffer)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkFramebuffer*)jniLongToPointer(framebuffer)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}



